src/consensus.rs

```diff
@@ -163,7 +164,8 @@ impl Consensus {
                     id,
                 },
             ))
-            .await?
+            .await
+            .context("Failed to add peer to known participant")?
```

Comment:
```suggestion
            .context("Failed to add peer to known participants")?
```

Reply:
I think I will drop `participant`.
The function is called only `add_peer_to_known` and the one below `add_peer_as_participant`.
It could be confusing.